test/test_irods_http_api.py

```diff
@@ -1825,6 +1825,332 @@ def test_non_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+
+            # Read the contents of the data object and show it contains exactly what we expect.
+            r = requests.get(self.url_endpoint, headers=headers, params={
+                'op': 'read',
+                'lpath': data_object,
+                'count': 30
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.content.decode('utf-8'), 'A' * 10 + 'B' * 10 + 'C' * 10)
+
+        finally:
+            # End the parallel write in case something failed.
+            # This avoids leaking of parallel-write resources in the HTTP API server.
+            if invoke_parallel_write_shutdown:
+                r = requests.post(self.url_endpoint, headers=headers, data={
+                    'op': 'parallel_write_shutdown',
+                    'parallel-write-handle': parallel_write_handle
+                })
+                self.logger.debug(r.content)
+
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+    def test_parallel_writes_using_header_based_form_with_data_exceeding_internal_write_threshold(self):
+        # This test assumes the HTTP API is configured to use a value smaller than
+        # 96kb for "/irods_client/max_number_of_bytes_per_write_operation". This is
+        # important because values greater than or equal to 96kb will result in the
+        # background thread pool only being used one time. This test is designed to
+        # force the HTTP API to schedule multiple writes across the thread pool.
+        #
+        # NOTE: This test DOES NOT and CANNOT confirm the background thread pool is
+        # being used because it does not have visibility into the server's internal
+        # mechanisms. Our only means of verification is by careful review of the log
+        # messages. The server is expected to output messages about multiple writes
+        # to the iRODS server. The thread ID should change from one write operation to
+        # the next.
+
+        headers = {'Authorization': f'Bearer {self.rodsuser_bearer_token}'}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/parallel_writes_header_based_1.txt'
+
+        # Generate 96kb string.
+        # This is the data we'll upload to the HTTP API.
+        data96kb = ('A' * 32768) + ('B' * 32768) + ('C' * 32768)
+
+        # Calculate a checksum for data.
+        # The checksum will be used to confirm the data was uploaded correctly.
+        checksum = base64.b64encode(hashlib.sha256(data96kb.encode('utf-8')).digest()).decode('utf-8')
```

Comment:
Not that we're super concerned about speed here, but since we already know the contents of the data object ahead of time, we can just hard-code the checksum here if we wanted to save some cycles.

Reply:
Very true. There are likely other instances of this in the test suite. I want to keep one around so it's easy to find. You never know when you might need to compute a checksum in python.

Will confirm there are other instances of this before updating.

Reply:
Done.